lib/url-pattern/url-match-pattern.js

```diff
@@ -64,19 +77,26 @@ _.assign(UrlMatchPattern.prototype, /** @lends UrlMatchPattern.prototype */ {
      */
     createMatchPattern: function () {
         var matchPattern = this.pattern,
+            match;
+
+        // Check the match pattern of sanity and split it into protocol, host and path
 
-            // Check the match pattern of sanity and split it into protocol, host and path
+        if (this._allowProtocols) {
+            match = matchPattern.match(regexes.noProtocolPatternSplit);
+        }
+        else {
             match = matchPattern.match(regexes.patternSplit);
+        }
 
         if (!match) {
             // This ensures it is a invalid match pattern
             return undefined;
         }
 
         return {
-            protocol: match[1],
-            host: match[2],
-            path: this.globPatternToRegexp(match[3])
+            protocol: this._allowProtocols ? null : match[1],
```

Comment:
Create whatever `var`s you need above and use them once. The `_allowProtocols` flag should only be used _once_ in the entire function, not in 4 separate branches (1 in the `if-else` and 3 ternary operators).

e.g: 
```
var protocol

if (_allowblabla) {
    protocol = something
}
else {
    protocol = somethingelse;
}

return { protocol: protocol, ....}
```